protocols/relay/src/v2.rs

```diff
@@ -42,48 +42,53 @@ pub mod client {
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::client::Transport instead."
+            note = "ClientTransport has been renamed to Transport
+            and is now private as it shouldn't be required by end users."
         )]
-        pub type ClientTransport = crate::client::Transport;
+        pub type ClientTransport = crate::priv_client::transport::Transport;
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::client::transport::Listener instead."
+            note = "RelayListener has been renamed to Listener
+            and is now private as it shouldn't be required by end users."
         )]
-        pub type RelayListener = crate::client::transport::Listener;
+        pub type RelayListener = crate::priv_client::transport::Listener;
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::transport::RelayedDial instead."
+            note = "RelayedDial is now private as it shouldn't be required by end users."
         )]
         pub type RelayedDial = BoxFuture<
             'static,
-            Result<crate::client::RelayedConnection, crate::client::transport::Error>,
+            Result<crate::client::RelayedConnection, crate::priv_client::transport::Error>,
         >;
 
         #[deprecated(
             since = "0.15.0",
-            note = "Use libp2p_relay::client::transport::Error instead."
+            note = "RelayError has been renamed to Error
+            and is now private as it shouldn't be required by end users."
```

Comment:
I'd say the error needs to remain public because - in theory - someone polling the `Transport` manually needs to be able to name it.

Reply:
I think you'll find that making the type declaration itself private is not going to work but making the entire module private and exporting select items from it does work!